src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -185,6 +181,7 @@ internal static bool IsDevice(StringBuilder path)
         /// <summary>
         /// Returns true if the path uses the extended syntax (\\?\)
         /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I've gone through them again. Validated via setting debug break points and perf tracing. Removed 3 and added 1.